tests/examples/CMakeLists.txt

```diff
@@ -310,11 +310,12 @@ foreach(file ${test_files})
     endif()
 endforeach()
 
-add_custom_target(generate-examples ALL DEPENDS ${EXAMPLE_DOC_GENERATED_FILES} ${EXAMPLE_DOC_EXTRA_TARGETS})
+add_custom_target(generate-examples ALL DEPENDS ${EXAMPLE_DOC_GENERATED_FILES})
 
 add_custom_target(check-examples
     ${CMAKE_COMMAND} ${CMAKE_SOURCE_DIR}
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    DEPENDS ${EXAMPLE_DOC_EXTRA_TARGETS})
```

Comment:
I feel like this should also use `add_custom_command` above (line 272, https://github.com/awesomeWM/awesome/pull/2545/files#diff-99add2d86fe5b8809426e4385f9e6403R272), and generate a "stamp file" / target for CMake to only run this when the input file changes.